### PR TITLE
gracefully handle missing savedrun for bot integration

### PR DESCRIPTION
**Pull request by nikochiko**
### Q/A checklist

- [ ] If you add new dependencies, did you update the lock file?
```bash
poetry lock --no-update
```
- [ ] Run tests 
```bash
ulimit -n unlimited && ./scripts/run-tests.sh
```
- [ ] Do a self code review of the changes - Read the diff at least twice.  
- [ ] Carefully think about the stuff that might break because of this change - this sounds obvious but it's easy to forget to do "Go to references" on each function you're changing and see if it's used in a way you didn't expect. 
- [ ] The relevant pages still run when you press submit
- [ ] The API for those pages still work (API tab)
- [ ] The public API interface doesn't change if you didn't want it to (check API tab > docs page)
- [ ] Do your UI changes (if applicable) look acceptable on mobile?
- [ ] Ensure you have not regressed the import time unless you have a good reason to do so. 
You can visualize this using tuna:
```bash
python3 -X importtime -c 'import server' 2> out.log && tuna out.log
```
To measure import time for a specific library:
```bash
$ time python -c 'import pandas'

________________________________________________________
Executed in    1.15 secs    fish           external
   usr time    2.22 secs   86.00 micros    2.22 secs
   sys time    0.72 secs  613.00 micros    0.72 secs
```
To reduce import times, import libraries that take a long time inside the functions that use them instead of at the top of the file:
```python
def my_function():
    import pandas as pd
    ...
```

### Legal Boilerplate

Look, I get it. The entity doing business as “Gooey.AI” and/or “Dara.network” was incorporated in the State of Delaware in 2020 as Dara Network Inc. and is gonna need some rights from me in order to utilize my contributions in this PR. So here's the deal: I retain all rights, title and interest in and to my contributions, and by keeping this boilerplate intact I confirm that Dara Network Inc can use, modify, copy, and redistribute my contributions, under its choice of terms.

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1208584814057545

### CODE REVIEW

File: daras_ai_v2/bots.py [hunk 1]
```diff
@@ -133,15 +131,8 @@ def __init__(self):
...
-            self.workspace = self.bi.workspace
-            self.current_user = self.bi.created_by
-        else:
-            # TODO: remove this once all bots have been migrated to workspaces
-            self.workspace = Workspace.objects.get_or_create_from_uid(
-                self.bi.billing_account_uid
-            )[0]
-            self.current_user = self.workspace.created_by
+        self.workspace = self.bi.workspace
+        self.current_user = self.bi.created_by
```
Comment (nikochiko) [inline daras_ai_v2/bots.py:135]: no longer needed.

i confirmed that there are no bot integrations that do not have a workspace

**Comment by nikochiko** [comment]
test failure is due to google translate

**Comment by nikochiko** [comment]
we need more code to fix this in the case of web chatbots. it doesn't error in 500 anymore but sends an empty 200 chunk and then 404 without the connected message (although it does reach that part and enter `bot.send_msg(PAGE_NOT_CONNECTED_ERROR)` code).

this does fix it for other integrations though and we should merge this first because it improves the current state of the codebase. i'll make another PR to fix the web integration case

**Comment by devxpy** [comment]
I think if we convert `bot.send_msg(PAGE_NOT_CONNECTED_ERROR)` to `raise UserError(PAGE_NOT_CONNECTED_ERROR)`, it will fix the web issue too?

**Review by devxpy — APPROVED** [review]

**Comment by nikochiko** [comment]
> I think if we convert bot.send_msg(PAGE_NOT_CONNECTED_ERROR) to raise UserError(PAGE_NOT_CONNECTED_ERROR), it will fix the web issue too?

this didn't work

**Pull request merged**